clients/unshaded/src/main/java/tachyon/client/TachyonFile.java

```diff
@@ -453,10 +453,20 @@ private TachyonByteBuffer readLocalByteBuffer(int blockIndex, long offset, long
    */
   TachyonByteBuffer readRemoteByteBuffer(ClientBlockInfo blockInfo) {
     // We call into the remote block in stream class to read a remote byte buffer
-    ByteBuffer buf =
-        RemoteBlockInStream.readRemoteByteBuffer(mTachyonFS, blockInfo, 0, blockInfo.length,
-            mTachyonConf);
-    return (buf == null) ? null : new TachyonByteBuffer(mTachyonFS, buf, blockInfo.blockId, -1);
+    // TODO: Avoid using this constructor, investigate the right way to read remote buffer here.
+    RemoteBlockInStream inStream = new RemoteBlockInStream(new TachyonFile(null, -1, null),
+        ReadType.NO_CACHE, -1, null, null, true);
+    ByteBuffer inBuf
+        = inStream.readRemoteByteBuffer(mTachyonFS, blockInfo, 0, blockInfo.length, mTachyonConf);
+    ByteBuffer outBuf = ByteBuffer.allocate((int) inBuf.capacity());
+    outBuf.put(inBuf);
+    try {
+      inStream.close();
```

Comment:
If the `readRemoteByteBuffer` or `allocate` fails then it will not. But then runtime errors will probably cause the client process to crash anyway. BTW, we should probably throw the IOException here instead of catching it (I'll fix that).